smithy-typescript-codegen-test/smithy-build.json

```diff
@@ -8,7 +8,8 @@
             "packageVersion": "0.0.1",
             "packageJson": {
                 "license": "Apache-2.0"
-            }
+            },
+            "memberNullabilityCompatibilityMode": "strict"
```

Comment:
I took some more time to think about this and discuss with @gosar. I think `requiredMemberMode` makes more sense, with enum values of `strict` and `nullable`.